spec/ics-25-handler-interface/README.md

```diff
@@ -0,0 +1,366 @@
+---
+ics: 25
+title: Handler Interface
+stage: draft
+category: ibc-core
+requires: 2, 3, 4, 23, 24
+required-by: 26
+author: Christopher Goes <cwgoes@tendermint.com>
+created: 2019-04-23
+modified: 2019-05-14
+---
+
+## Synopsis
+
+This document describes the interface exposed by the standard IBC implementation (referred to as the IBC handler) to modules within the same state machine, and the implementation of that interface by the IBC handler.
+
+### Motivation
+
+IBC is an inter-module communication protocol, designed to faciliate reliable, authentication message passing between modules on separate blockchains. Modules should be able to reason about the interface they interact with and the requirements they must adhere to in order to utilize it safely.
+
+### Definitions
+
+`Client` and `ConsensusState` are as defined in ICS 2.
+
+`Connection` and `ConnectionState` are as defined in ICS 3.
+
+`Channel` and `Packet` are as defined in ICS 4.
+
+`CommitmentProof` is as defined in ICS 23.
+
+`Identifier`s must conform to the schema defined in ICS 24.
+
+### Desired Properties
+
+- Creation of clients, connections, and channels should be as permissionless as possible.
+- The module set should be dynamic: chains should be able to add and destroy modules at will with a persistent IBC handler.
+- Modules should be able to write their own more complex abstractions on top of IBC to provide additional semantics or guarantees.
+
+## Specification
+
+### Clients
+
+By default, clients are unowned: any module can create a new client, query any existing client, update any existing client, and delete any existing client not in use.
+
+`ClientKind` enumerates the list of light client algorithms supported by the handler implementation.
+
+```golang
+type ClientKind enum {
+  Tendermint
+}
+```
+
+`ClientOptions` contains all the parameter choices required to create a client.
+
+```golang
+type ClientOptions struct {
```

Comment:
Is it needed to differentiate `ClientOptions` and `ClientInfo`?

Reply:
Ah not really, this part still needs a bit of cleanup to match ICS 4